SharpPcap/DeviceConfiguration.cs

```diff
@@ -14,28 +14,28 @@
 You should have received a copy of the GNU Lesser General Public License
 along with SharpPcap.  If not, see <http://www.gnu.org/licenses/>.
 */
-/* 
- * Copyright 2017 Noah Potash <noah.potash@outbreaklabs.com>
+/*
+ * Copyright 2010 Chris Morgan <chmorgan@gmail.com>
  */
 
-
 namespace SharpPcap
 {
-    /// <summary>
-    /// The activation of monitor mode used when opening a device
-    /// </summary>
-    public enum MonitorMode : short
+    public class DeviceConfiguration
     {
         /// <summary>
-        /// Monitor mode.
-        /// Allows capturing of 802.11 wireless packets even when not associated
-        /// with a network.
+        /// Defines if the adapter mode. 
         /// </summary>
-        Active = 1,
+        public DeviceModes Mode { get; set; }
```

Comment:
@chmorgan maybe we should replace the enum with boolean properties?

Reply:
@kayoub5 I'd typically look at the usage to determine if thats an option. In many cases a boolean might be how it works internally but when you read the code you end up with something unclear, like somefunctioncall(true); which is pretty opaque compared to say somefuntioncall(CaptureWireless); That's why you'll see heavy usage of binary enums, because true and false just don't provide enough context when reading the code.

if we had an enum like "Normal" and "CaptureRawWireless" that might help with providing the user with info about why you might use it. It looks like that mode is pretty specific to raw 802.11 capture which helps.

Thoughts?

Reply:
Just to make sure we are on the same page, I am talking about `DeviceModes` (flags), I feel you are talking about `MonitorMode`.

Reply:
oh do you mean that we should consider breaking out the DeviceModes flags into individual booleans where each flag is a boolean in the DeviceConfiguration?

Reply:
Yes

Reply:
Would it help with usability? If so then sure, lets see how it looks in usage. Maybe a follow up PR after this one is merged?

I was talking monitor modes before btw but think those comments still have merit for monitor modes re: enum vs boolean.

Reply:
* Monitor mode: enum restored
* DeviceModes: will keep the enum for now